controllers/v1/reportsController.js

```diff
@@ -3012,4 +3017,238 @@ module.exports = class Reports {
+          }
+          surveySubmissionsDocument['programInfo'] = programDocument[0];
+        }
+        let data = surveySubmissionsDocument;
+
+        let transformedData = transformData(data);
+
+        let chartData = await helperFunc.instanceReportChart(transformedData, filesHelper.survey);
+        chartData.solutionName = data.solutionExternalId;
+
+        let surveyAnswers = data.answers;
+
+        let evidenceData = formateEvidenceData(surveyAnswers, chartData);
+
+        let responseObj;
+
+        if (evidenceData.length > 0) {
+          responseObj = await helperFunc.evidenceChartObjectCreation(
+            chartData,
+            evidenceData
+          );
+        } else {
+          responseObj = chartData;
+        }
+
+        return resolve({
+          status: httpStatusCode.ok.status,
+          message: responseObj.response,
+        });
+      }
+    } catch (err) {
+
+      let response = {
+        result: false,
+        message: err.message || 'INTERNAL_SERVER_ERROR',
+      };
+      return resolve(response);
+    }
+  });
+};
+
 };
+
+function transformData(data) {
+
+  //custom function return to mimic druid response from data source
+  let singleQuestionAnswerDataPerSolutionArr = [];
+
+  let solutionExternalId = data.solutionExternalId;
+
+  let solutionId = data.solutionId;
+  let solutionName = data.solutionInfo.name;
+
+  let surveyId = data.surveyId;
+  let surveySubmissionId = data._id;
+  let createdAt = data.createdAt;
+  let completedAt = data.completedDate;
+  let createdBy = data.createdBy;
+
+  let criteriaExternalId = data.criteria[0].externalId;
+  let criteriaId = data.criteria[0]._id;
+  let criteriaName = data.criteria[0].name;
+  let isAPrivateProgram = data.isAPrivateProgram;
+
+  let answersObject = data.answers;
+
+  let answerKeys = Object.keys(answersObject);
+
+  answerKeys.forEach((key) => {
+    let singleAnswerObj = answersObject[key];
+
+    singleQuestionAnswerDataPerSolutionArr.push({
+      event: {
+        solutionExternalId,
+        solutionId,
+        solutionName,
+        surveyId,
+        surveySubmissionId,
+        createdAt,
+        completedAt,
+        createdBy,
+        criteriaExternalId,
+        criteriaId,
+        criteriaName,
+        isAPrivateProgram,
+        questionAnswer: singleAnswerObj.value[0],
+        questionECM: singleAnswerObj.evidenceMethod,
+        questionExternalId: singleAnswerObj.externalId,
+        questionId: singleAnswerObj.qid,
+        questionName: singleAnswerObj.question[0],
+        questionResponseLabel: singleAnswerObj.value,
+        //questionResponseLabel_number:,
+        questionResponseType: singleAnswerObj.responseType,
+        // question_response_number
+        answerOptions: singleAnswerObj.options ? singleAnswerObj.options : undefined,
+      },
+    });
+  });
+
+  return singleQuestionAnswerDataPerSolutionArr;
+  //return data;
+}
+
+function formateEvidenceData(surveyAnswers, chartData) {
```

Comment:
@borkarsaish65 why we are adding these functions in controller? Are these new endpoints? Controller should call module helper and get the data and return the response. All other logics we should not have in controllers

Reply:
changed now @VISHNUDAS-tunerlabs

Reply:
not used anymore so didnt make changes